handler.go

```diff
@@ -332,11 +333,36 @@ func (ca *Cagent) GetAllMeasurements() (MeasurementsMap, error) {
 
 	if len(errs) == 0 {
 		measurements["cagent.success"] = 1
+	} else {
+		measurements["cagent.success"] = 0
+	}
 
-		return measurements, nil
+	// measurements fetched below should not affect cagent.success
+	ca.smartInit.Do(func() {
```

Comment:
In all other checks, they were checking for feature availability in its implementation. (And returning nil result in case if some package is not installed or feature disabled by cagent cofiguration). I think this check and smartmon parsing should be moved to some function that will abstract this checks. Currently, the `GetAllMeasurements` function became even more inconsistent.

Reply:
I would leave as is till common approach for `GetAllMeasurements` found

Reply:
I agree with @nikita-vanyasin 
The function is already quite big but so far it did not do checking if things are available or not. If something is not available or caused an error, the underlying function returns an empty struct, not `nil`, and the errors. I think we should continue like this

Reply:
done